calico_containers/adapter/datastore.py

```diff
@@ -444,36 +463,112 @@ def get_bgp_peers(self, version):
         Get the configured BGP Peers
 
         :param version: "v4" for IPv4, "v6" for IPv6
-        :return: List of netaddr.IPAddress IP addresses.
+        :return: List of dictionaries:
```

Comment:
Can we avoid passing dictionaries around like this?  I'd much rather see an object (maybe based on `NamedTuple`) to represent this data.  It makes it much easier to catch bugs since you can't set arbitrary keys, can assert on types.

Also, we'd be able to reuse it for the node-specific stuff.  C.f. `Profile` or `Rule` objects in this file.